src/libstd/sync/condvar.rs

```diff
@@ -60,6 +60,10 @@ impl WaitTimeoutResult {
     ///         break
     ///     }
     /// }
+    /// // Leaving a condition variable's mutex held after we're no longer
+    /// // watching it will block and possibly deadlock notifier threads, so
```

Comment:
I think this is misleading. Just holding the mutex would not be sufficient to block a notifier thread. It just stops a notifier thread (or anything else) from acquiring the same mutex, which is the whole point of a mutex. If I understand correctly, the `notify_one` call does not block, so there is nothing about the behavior that would be specific to a notifier thread.